src/webrtc/callFeed.ts

```diff
@@ -237,17 +286,36 @@ export class CallFeed extends TypedEventEmitter<CallFeedEvent, EventHandlerMap>
 
         this.analyser.getFloatFrequencyData(this.frequencyBinCount);
 
-        let maxVolume = -Infinity;
+        let maxCurrentVolume = -Infinity;
         for (let i = 0; i < this.frequencyBinCount.length; i++) {
-            if (this.frequencyBinCount[i] > maxVolume) {
-                maxVolume = this.frequencyBinCount[i];
+            if (this.frequencyBinCount[i] > maxCurrentVolume) {
+                maxCurrentVolume = this.frequencyBinCount[i];
             }
         }
-
+        this.maxVolume = maxCurrentVolume;
         this.speakingVolumeSamples.shift();
-        this.speakingVolumeSamples.push(maxVolume);
-
-        this.emit(CallFeedEvent.VolumeChanged, maxVolume);
+        this.speakingVolumeSamples.push(this.maxVolume);
+
+        this.emit(CallFeedEvent.VolumeChanged, this.maxVolume);
+
+        // Handle voice activity detection only if it is enabled and user has not manually muted themselves
+        if (this.VADEnabled) {
+            // If the user is speaking
+            if (this.maxVolume > this.voiceActivityThreshold) {
+                this.VADCooldownStarted = new Date();
```

Comment:
Would it make more sense to set the cooldown when the user stops speaking? It somehow feels better to only mute the user if we have seen they did stop speaking

Reply:
It is not much different, since it will still be called every time the user is below threshold but I changed it.